src/lib.rs

```diff
@@ -37,7 +37,7 @@ where
     /// The detected frontmatter, if any.
     pub frontmatter: Option<Frontmatter<'a>>,
     source: T,
-    state: DocumentAttributeParserState<'a>,
+    pub state: DocumentAttributeParserState<'a>,
```

Comment:
Why was this made public? It seems like the new function adds the desired functionality without leaking implementation details.

Reply:
If you didn't want to edit the crate itself, making this public would allow you to take a FrontmatterExtractor and iterate over it until the crate believe it's done. Not really required with the addition of iterate_and_return(), but it could allow more flexibility in the future.

I can revert that if you'd like